storage/postgres/client.go

```diff
@@ -250,9 +250,32 @@ func (c *Client) Wipe(ctx context.Context) error {
 		}
 	}
 
+	// list, then drop some extensions.
+	rows, err := c.Query(ctx, `
+		SELECT extname
+		FROM pg_extension
+	`)
+	if err != nil {
+		return fmt.Errorf("failed to list extensions: %w", err)
+	}
+	defer rows.Close()
```

Comment:
we use multiple `defer rows.Close()` in this function, reassigning `rows` :skull: :skull: :skull:

Reply:
I think `defer` captures enough context that it's all OK? Example: https://go.dev/play/p/6g8xQemALDi

If not, please do fix :pray:

Reply:
struct, pointer receiver
https://go.dev/play/p/fWui7UAvo7N -> 2 2 :x: 

pointer to struct, pointer receiver
https://go.dev/play/p/FNsKXdCQ7Fs -> 2 1 :heavy_check_mark: 

interface with pointer to struct, pointer receiver
https://go.dev/play/p/Nywd7MMYyvd -> 2 1 :heavy_check_mark: 

interface with struct, struct receiver
https://go.dev/play/p/E4f6lT8zO5o -> 2 1 :heavy_check_mark:

Reply:
this stuff is too... it's like the phrase "sensitive to initial conditions" that you might say of a physical system. I'm just gonna wrap these in iifes.

Reply:
although I'm going to get rid of this and go with the create extension if not exists way

Reply:
hm there's a listIndexerTables in its own method, so I'll go with methods